docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,210 @@
+# ZONE11: SPF policy validation
+
+## Test case identifier
+**ZONE11**
+
+## Table of contents
+
+* [Objective](#objective)
+* [Scope](#scope)
+* [Inputs](#inputs)
+* [Summary](#summary)
+* [Test procedure](#test-procedure)
+* [Outcome(s)](#outcomes)
+* [Special procedural requirements](#special-procedural-requirements)
+* [Intercase dependencies](#intercase-dependencies)
+* [Terminology](#terminology)
+
+## Objective
+
+Sender Policy Framework (SPF), described in [RFC 7208], is a mechanism
+allowing domain name owners to specify which hosts are allowed to send mail
+claiming to be from that domain. It is implemented by means of TXT records in a
+structured format.
+
+This test case looks up SPF records in the apex of *Child Zone*. It checks
+that there is at most one published SPF version 1 policy and, if present, also
+checks its syntax.
+
+## Scope
+
+It is assumed that *Child Zone* has been tested and reported by
+[Connectivity01]. This test case will just ignore non-responsive name servers
+or name servers not giving a correct DNS response for an authoritative name
+server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                      | Level   | Arguments    | Message ID for message tag
+:--------------------------------|:--------|:-------------|:--------------------------------------------
+Z11_INCONSISTENT_SPF_POLICIES    | WARNING |              | The *Child Zone* publishes different SPF policies on different name servers.
+Z11_DIFFERENT_SPF_POLICIES_FOUND | NOTICE  | ns_ip_list   | The following name servers returned the same SPF version 1 policy, but other name servers returned a different policy. Name servers: {ns_ip_list}.
+Z11_NO_SPF_FOUND                 | INFO    |              | The *Child Zone* does not publish an SPF policy.
```

Comment:
Looking at this again, is it a problem that SPF policy is missing? Is that against recommendations? Should we raise the level to NOTICE when there is no policy?

Do you see a connection between the existence of MX and existence of SPF for a domain?

Reply:
IMHO, not publishing an SPF policy is similar to not signing a zone with DNSSEC: not an error in itself, but maybe something a domain name owner should consider implementing.

According to my data, on the `.fr` zone, 62.6% of zones have both MX records and an SPF policy, and 33.2% of zones have no MX at all but an SPF policy. I assume most of these would be “null SPF” policies (i.e. `v=spf -all`).

The same can be said for DKIM (unless the SPF policy is a so-called “null SPF”, such as `v=spf1 -all`) and DMARC.

Reply:
In order to be consistent with the level of the message Zonemaster outputs if there is no DNSSEC signature, I think we can raise the message level for `Z11_NO_SPF_FOUND` to NOTICE.

Regardless of whether the domain originates mail, it is a good practice to implement SPF:
* if a domain originates mail, specifying the mail servers authorized to send mail on the domain’s behalf is a basic defense against spoofing and even though SPF alone won’t stop all forms of spoofing, it’s required by DMARC;
* if a domain does not originate mail (for example, when it’s a parked domain), then it’s a [good practice](https://www.m3aawg.org/sites/default/files/m3aawg_parked_domains_bcp-2022-06.pdf) to publish a “null SPF” policy.

Reply:
>     * if a domain does not originate mail (for example, when it’s a parked domain), then it’s a [good practice](https://www.m3aawg.org/sites/default/files/m3aawg_parked_domains_bcp-2022-06.pdf) to publish a “null SPF” policy.

In ZONE09 the test case detects Null MX. Is it easy to detect a Null SPF? Should this test case report Null SPF and update ZONE09 to report Null MX?